csf_tz/csftz_hooks/payment_entry.py

```diff
@@ -150,3 +150,23 @@ def get_outstanding_reference_documents(args):
         )
 
     return data
+
+def validate(self, method):
+    restrict_unallocated_amount_for_supplier()
+
+# Check Feature Flag in Company Settings
+def restrict_unallocated_amount_for_supplier():
+    company = frappe.get_doc("Company", frappe.defaults.get_user_default("company"))
```

Comment:
1. Use self.company and not get_user_default
2. Use get_cached_doc or get_cached_value